frontend/src/metabase/admin/databases/components/DeleteDatabaseModel/DeleteDatabaseModal.tsx

```diff
@@ -0,0 +1,161 @@
+}
+
+const DeleteDatabaseModal = ({
+  database,
+  onClose,
+  onDelete,
+}: DeleteDatabaseModalProps) => {
+  const { data: usageInfo } = useFetch<DatabaseUsageInfo>(
+    `/api/database/${database.id}/usage_info`,
+  );
+
+  const [isContentRemovalConfirmed, setIsContentRemovalConfirmed] =
+    useState(false);
+  const [databaseNameConfirmation, setDatabaseNameConfirmation] = useState("");
+  const [error, setError] = useState<any>(null);
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+
+    try {
+      onDelete(database);
+      // immediately call on close because database deletion should be non blocking
+      onClose();
+    } catch (error) {
+      setError(error);
+    }
+  };
+
+  const handleEditConnectionDetailsClick: React.MouseEventHandler = e => {
+    e.preventDefault();
+    onClose();
+  };
+
+  const hasContent = usageInfo && hasContentInDatabase(usageInfo);
+
+  const isDatabaseNameConfirmed =
+    databaseNameConfirmation.trim().toLowerCase() ===
+    database.name.trim().toLowerCase();
+  const shouldShowDbNameInputConfirmation =
+    isContentRemovalConfirmed || !hasContent;
+  const canDelete =
+    (isContentRemovalConfirmed || !hasContent) && isDatabaseNameConfirmed;
+
+  const deleteButtonLabel = hasContent
+    ? t`Delete this content and the DB connection`
+    : t`Delete`;
+
+  const errorMessage = getErrorMessage(error);
+
+  return (
+    <ModalContent
+      title={t`Delete the ${database.name} database?`}
+      onClose={onClose}
+    >
+      <DeleteDatabaseModalRoot onSubmit={canDelete ? handleSubmit : undefined}>
+        {!shouldShowDbNameInputConfirmation && (
+          <DeleteDatabaseModalSection>
+            <Alert icon="info">
+              {jt`If you’re trying to migrate from a development DB to a production one, you don’t need to do this. You can just ${(
+                <Button
+                  onlyText
+                  onClick={handleEditConnectionDetailsClick}
+                >{`edit your connection details.`}</Button>
+              )}`}
+            </Alert>
+          </DeleteDatabaseModalSection>
+        )}
+        {hasContent && (
+          <>
+            <DeleteDatabaseModalSection>
+              {t`Deleting this database will also delete everything based on it. If you’re really trying to do this, please check each of these boxes:`}
+            </DeleteDatabaseModalSection>
+
+            <DeleteDatabaseModalSection>
+              <ContentRemovalConfirmation
+                usageInfo={usageInfo}
+                onChange={setIsContentRemovalConfirmed}
+              />
+            </DeleteDatabaseModalSection>
+          </>
+        )}
+        {shouldShowDbNameInputConfirmation && (
+          <>
+            <DeleteDatabaseModalSection>
+              <Alert icon="warning" variant="error">
+                {t`This will delete every saved question, model, metric, and segment you’ve made that uses this data, and can’t be undone! `}
```

Comment:
I replaced the line below because technically it is correct to use `’`. Actually, there are plenty of occurrences of `’` in our codebase so I'll stick to it here